test/unit/test_session_manager.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All right reserved.
+#
+
+from typing import Optional
+from unittest import mock
+
+import pytest
+
+from snowflake.connector.network import SnowflakeRestful
+
+pytestmark = pytest.mark.skipolddriver
+
+hostname_1 = "sfctest0.snowflakecomputing.com"
+url_1 = f"https://{hostname_1}:443/session/v1/login-request"
+
+hostname_2 = "sfc-ds2-customer-stage.s3.amazonaws.com"
+url_2 = f"https://{hostname_2}/rgm1-s-sfctest0/stages/"
+url_3 = f"https://{hostname_2}/rgm1-s-sfctst0/stages/another-url"
+
+
+def get_mock_connection():
```

Comment:
```suggestion
def get_mock_connection() -> None:
```

Reply:
But I actually think that this doesn't have to be a function, but the stuff this fn returns could be a module level variable